File: TS/Prompt.ts
/**
 * A prompt or prompt template that the server offers.
 */
import { PromptArgument } from './PromptArgument';

export interface Prompt {
  /**
   * The name of the prompt or prompt template.
   */
  name: string;
  
  /**
   * An optional human-readable description of the prompt.
   */
  description?: string;
  
  /**
   * The arguments that this prompt template accepts, if any.
   */
  arguments?: PromptArgument[];
}

File: Sources/Schema/Prompt.swift
/**
 * A prompt or prompt template that the server offers.
 */
public struct Prompt: Codable {
    /**
     * The name of the prompt or prompt template.
     */
    public var name: String
    
    /**
     * An optional human-readable description of the prompt.
     */
    public var description: String?
    
    /**
     * The arguments that this prompt template accepts, if any.
     */
    public var arguments: [PromptArgument]?
    
    public init(name: String, description: String? = nil, arguments: [PromptArgument]? = nil) {
        self.name = name
        self.description = description
        self.arguments = arguments
    }
}
